core/src/main/scala/com/clairvoyant/restonomer/core/authentication/RestonomerAuthentication.scala

```diff
@@ -54,6 +56,57 @@ case class BasicAuthentication(
 
 }
 
+case class JwtAuthentication(
+    subject: Option[String] = None,
+    secretKey: Option[String] = None,
+    algo: Option[String] = None
+) extends RestonomerAuthentication {
```

Comment:
Subject and Secret Key should be mandatory as this is the minimum requirement for authentication. Hence type should not be Option.
Algo can be optional because if the user doesn't provide the same, then we would consider HS256 as default.